packages/jest-haste-map/src/crawlers/watchman.js

```diff
@@ -28,15 +28,20 @@ function WatchmanError(error: Error): Error {
 module.exports = async function watchmanCrawl(
   options: CrawlerOptions,
 ): Promise<InternalHasteMap> {
+  const fields = ['name', 'exists', 'mtime_ms'];
   const {data, extensions, ignore, roots} = options;
   const defaultWatchExpression = [
     'allof',
     ['type', 'f'],
     ['anyof'].concat(extensions.map(extension => ['suffix', extension])),
   ];
   const clocks = data.clocks;
-
   const client = new watchman.Client();
+
+  if (options.sha1) {
+    fields.push('content.sha1hex');
```

Comment:
Don't we need to do a capability check with watchman? We need to stay backwards compatible here. To test, you can probably just use any property that doesn't exist and  see how watchman behaves. Watchman also has a capability check function to test whether this will work or not, we should be safe and fall back and calculate the sha1 separately like we do for the node crawler if that happens.

Reply:
Maybe the field specification is already ignored if it doesn't exist on watchman side? For robustness, I'd expect it to crash though, so I agree with Christoph a check (either capabilities, or perhaps just version number) is necessary to ensure we stay backward compatible.

We probably want to stay backward compatible because some companies may use older versions of Watchman with React Native, and may not be able to upgrade readily (for unrelated concerns, ex. compatibility with their source control).

Reply:
I added a specific check for that by using `list-capabilities`.